design-documents/graph-ql/coverage/returns.md

```diff
@@ -0,0 +1,348 @@
+## Configuration 
+
+The following settings should be accessible via `storeConfig` query:
+- Returns functionality status on the storefront: enabled/disabled
+
+Scenarios which may need these settings include:
+- Rendering of the returns section in the customer account
+
+```graphql
+{
+  storeConfig {
+    sales_magento_rma_enabled
+  }
+}
+```
+
+## Use cases
+
+### View return list with pagination in customer account
+
+```graphql
+{
+  customer {
+    returns(pageSize: 10, currentPage: 2) {
+      items {
+        id
+        creation_date
+        customer_name
+        status
+      }
+      page_info {
+        current_page
+        page_size
+        total_pages
+      }
+      total_count
+    }
+  }
+}
+```
+
+### View return list with pagination in order details
+
+```graphql
+{
+  customer {
+    orders(filter: {number: {eq: "00000008"}}) {
+      items {
+        returns(pageSize: 10, currentPage: 1) {
+          items {
+            id
+            creation_date
+            customer_name
+            status
+          }
+          page_info {
+            current_page
+            page_size
+            total_pages
+          }
+          total_count
+        }
+      }
+    }
+  }
+}
+```
+
+### View return details
+
+```graphql
+{
+  customer {
+    return(id: "0000003") {
+      id
+      order_id
+      creation_date
+      customer_email
+      customer_name
+      status
+      comments {
+        id
+        text
+        created_at
+        created_by
+      }
+      items {
+        id
+        product {
+          sku
+          name
+        }
+        custom_attributes {
+          id
+          label
+          value
+        }
+        request_quantity
+        quantity
+        status
+      }
+      shipping {
+        tracking {
+          id
+          carier {
+            label
+          }
+          tracking_number
+        }
+        address {
+          contact_name
+          street
+          city
+          region {
+            code
+          }
+          country {
+            full_name_english
+          }
+          postcode
+          telephone
+        }
+      }
+    }
+  }
+}
+```
+
+### Create a return
+
+#### Determine whether any order items are eligible for return
+
+There is a need to know if any order items are eligible for return. In the Luma example this is dictating whether "Return" link will be displayed on the order details page.
+
+```graphql
+{
+  customer {
+    orders(filter: {number: {eq: "00000008"}}) {
+      items {
+        items_eligible_for_return {
+          id
+          product_name
+        }
+      }
+    }
+  }
+}
+```
+
+#### Render return form with dynamic RMA attributes
+
+```graphql
+{
+  pageSpecificCustomAttributes(page_type: RETURN_ITEM_EDIT_FORM) {
```

Comment:
Can you explain this scenario in terms of the UI it's meant to support? It's not entirely clear to me right now how the UI is intended to use this data.

Reply:
We need to know which attributes need to be shown on the specific page. Not all EAV attributes of return items should be shown on return creation page, for example. 
Admin user can create new attributes and mark them as available for return creation form.

Additional examples are mentioned in comment for other entities:
```
    # See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/framework/attributes-metadata.md
    # PRODUCTS_COMPARE
    # PRODUCTS_LISTING
    # ADVANCED_CATALOG_SEARCH
```